### PR TITLE
Prevent node crashing when logger fails to serialize BigInt by using string interpolation

### PR DESCRIPTION
**Pull request by Vovans**
Addressing https://github.com/iron-fish/ironfish/issues/507 
The problem was with a logger which unsuccessfully tried to serialize an object with a BigInt value (newBlock.header.target). This was the cause of a node crash. It would be possible to fix this by overriding the JSON.stringify, but it is deep down in js-message library, so an easier fix is just to interpolate the value to a string since it is only used for debugging purposes.

Maybe it is a good idea to wrap the InterceptReporter.log with a try/catch, since it is really not that good if logging crashes things.

### CODE REVIEW

**Review by dguenther — CHANGES_REQUESTED** [review]
Thanks for looking at this, it'd be great to get this fixed! I think your comments are alluding to this, but it'd be nice to make this a more comprehensive fix in case we ever add more bigint logs in the future.

We have an extended JSON serializer in the codebase here to handle buffers and bigints: https://github.com/iron-fish/ironfish/blob/b2a53668e784ffeac1ea3fa9b7f6e5a0370064e8/ironfish/src/serde/iJson.ts#L11

I think if we call that on `args`, it would behave similarly to `console.log`, but avoid the error: https://github.com/iron-fish/ironfish/blob/b2a53668e784ffeac1ea3fa9b7f6e5a0370064e8/ironfish/src/rpc/routes/node/getLogStream.ts#L44

Then this test will need to be updated as well: https://github.com/iron-fish/ironfish/blob/b2a53668e784ffeac1ea3fa9b7f6e5a0370064e8/ironfish/src/rpc/routes/node/getLogStream.test.ts#L28

I'm open to considering other serialization strategies as well if they produce results similar to `console.log`. Thanks for looking at this!

**Comment by Vovans** [comment]
Sorry, the last push was a mistake. I'll change it in a bit.

**Comment by Vovans** [comment]
So the node is not getting crashed because of the problem in js-message library (it uses JSON.stringify, which can't handle bigints). Now we are serializing logObj.args before that. 

Additionally, we improve logs a bit. As was mentioned in this issue https://github.com/iron-fish/ironfish/issues/776, it was unexpected "\n"'s.

**Comment by NullSoldier** [comment]
Can you rebase this PR and get the lint to pass before we review it? This fix is valuable and I don't want to steal it from you.

**Comment by Vovans** [comment]
Rebased, and lint is passed

**Comment by dguenther** [comment]
Sorry for the long response time on this over the holidays! I'm going to take over getting this merged today.

File: ironfish-cli/src/commands/logs.ts [hunk 1]
```diff
@@ -23,11 +23,19 @@ export default class LogsCommand extends IronfishCommand {
     const response = this.sdk.client.getLogStream()
 
     for await (const value of response.contentStream()) {
+      let parsedArgs
+      try {
+        parsedArgs = IJSON.parse(value.args) as unknown[]
+      } catch {
+        this.logger.error(`Failed to deserialize args: ${value.args}`)
+        continue
+      }
```
Comment (NullSoldier) [inline ironfish-cli/src/commands/logs.ts:32]: Should we let this crash, or render the error? So users can send it to us? These will be things we need to fix, right?

I'd like to either let it crash, or just use `ErrorUtils.renderError(error, true)`

Comment (dguenther) [reply ironfish-cli/src/commands/logs.ts:32]: I'll re-throw the error here so it crashes

**Review by dguenther — APPROVED** [review]

**Pull request merged**